### PR TITLE
feat: add docs deploy index scraping actions for pymeilisearch

### LINKED ISSUE #294: Implement doc-deploy-index actions
💡 **Idea**
With [pymeilisearch](https://github.com/ansys/pymeilisearch) public, it is possible now for projects to upload indexed documentation.

Two actions should be implemented: `doc-deploy-dev-index` and `doc-deploy-stable-index`. These actions should be used in the following manner:

```yml
- name: "Doc deploy dev index"
  uses: ansys/actions/doc-deploy-dev-index@v4
    with:
      - doc-artifact-name: ...
      - cname: ...
      - library-name: ...             # Extract version number from here
      - host-url: ...
      - api-key: ...
```

### PR DESCRIPTION
**Pull request by Revathyvenugopal162**
Resolves #294

### CODE REVIEW

File: build-library/action.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ inputs:
 
   python-version:
     description: >
-      Python version used for installing and running ``pre-commit``.
+      Python version used for installing and execution of the build process.
```
Comment (jorgepiloto) [inline build-library/action.yml:31]:
```suggestion
      Python version used for installing ``build`` library.
```

File: build-wheelhouse/action.yml [hunk 2]
```diff
@@ -27,7 +27,7 @@ inputs:
 
   python-version:
     description: >
-      Python version used for installing and running ``pre-commit``.
+      Python version used for installing and execution of the build wheel process.
```
Comment (jorgepiloto) [inline build-wheelhouse/action.yml:30]:
```suggestion
      Python version used for installing and generating the wheelhouse artifact.
```

File: check-licenses/action.yml [hunk 3]
```diff
@@ -48,7 +48,7 @@ inputs:
 
   python-version:
     description: >
-      Python version used for installing and running ``pre-commit``.
+      Python version used for installing and executing licence check.
```
Comment (jorgepiloto) [inline check-licenses/action.yml:51]:
```suggestion
      Python version used for installing ``pip-licenses``.
```

File: doc-deploy-dev-index/action.yml [hunk 4]
```diff
@@ -0,0 +1,139 @@
+name: >
+  Doc deploy dev index action
+
+description: |
+  This action automates the process of creating indexes and scraping the
+  desired HTML documentation artifact that contains the development version
+  of a library, then deploying it to a Meilisearch instance.
```
Comment (jorgepiloto) [inline doc-deploy-dev-index/action.yml:7]:
```suggestion
  This action automates the process of scraping and creating indexes from the
  desired development documentation artifacts in HTML form. Later, the action
  deploys the index to a Meilisearch instance.
```

File: doc-deploy-dev-index/action.yml [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        else
+          echo "REPOSITORY=${{ inputs.repository }}" >> $GITHUB_ENV
+        fi
+
+    - name: "Checkout ${{ env.REPOSITORY }} repository"
+      uses: actions/checkout@v3
+      with:
+        repository: ${{ env.REPOSITORY }}
+        ref: ${{ inputs.branch }}
+        token: ${{ inputs.token }}
```
Comment (jorgepiloto) [inline doc-deploy-dev-index/action.yml:86]: Why checking the repository? No need to clone the repo. We should only install ``pymeilisearch`` and upload the indexed artifacts to the instance.

Comment (jorgepiloto) [inline doc-deploy-dev-index/action.yml:86]:
```suggestion
```

File: doc-deploy-dev-index/action.yml [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+    - uses: ansys/actions/_logging@main
+      with:
+        level: "INFO"
+        message: >
+          Install the pymeilisearch required for the indexing.
+
+    - name: "Set up Python"
+      uses: ansys/actions/_setup-python@main
+      with:
+        python-version: '3.10'
```
Comment (jorgepiloto) [inline doc-deploy-dev-index/action.yml:118]:
```suggestion
        python-version: ${{ inputs.python-version }}
```

File: doc-deploy-dev-index/action.yml [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+    - name: "Install build and twine"
```
Comment (jorgepiloto) [inline doc-deploy-dev-index/action.yml:120]:
```suggestion
    - name: "Install pymeilisearch"
```

File: doc-deploy-dev-index/action.yml [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+    - uses: ansys/actions/_logging@main
+      with:
+        level: "INFO"
+        message: >
+          Scrap the document and deploy it to pymeilisearch.
+
+    - name: Scrape the dev documentation to meilisearch
+      shell: bash
+      run: |
+        pymeilisearch upload --template ${{ inputs.template }} --index ${{ inputs.library_name }}-vdev  --cname ${{ inputs.cname }}/version/dev/ html ${{ inputs.doc-artifact-name }}
```
Comment (jorgepiloto) [inline doc-deploy-dev-index/action.yml:136]: No need to add `v` in front of the `dev` suffix:

```suggestion
        pymeilisearch upload --template ${{ inputs.template }} --index ${{ inputs.library-name }}-dev  --cname ${{ inputs.cname }}/version/dev/ html ${{ inputs.doc-artifact-name }}
```

File: doc-deploy-stable-index/actions.yml [hunk 9]
```diff
@@ -0,0 +1,195 @@
+name: >
+  Doc deploy stable index action
+
+description: |
+  This action automates the process of creating indexes and scraping the
+  stable HTML documentation artifact that contains the development version
+  of a library, then deploying it to a Meilisearch instance.
```
Comment (jorgepiloto) [inline doc-deploy-stable-index/actions.yml:7]:
```suggestion
  This action automates the process of scraping and creating indexes from the
  desired stable documentation artifacts in HTML form. Later, the action
  deploys the index to a Meilisearch instance.
```

File: doc-deploy-stable-index/actions.yml [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+    required: false
+    default: "sphinx_pydata"
+    type: string
+
+  target:
+    description: >
+      Optional target used during the building process for version.
+    required: false
+    default: ''
+    type: string
```
Comment (jorgepiloto) [inline doc-deploy-stable-index/actions.yml:69]: I can see the issue with the target. To simplify these actions, let's allow users to pass the name of the index. This simplifies a lot our action. This applies also to the dev action.

File: doc-deploy-stable-index/actions.yml [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+          echo "REPOSITORY=${{ inputs.repository }}" >> $GITHUB_ENV
+        fi
+
+    - name: "Checkout ${{ env.REPOSITORY }} repository"
+      uses: actions/checkout@v3
+      with:
+        repository: ${{ env.REPOSITORY }}
+        ref: ${{ inputs.branch }}
+        token: ${{ inputs.token }}
+
```
Comment (jorgepiloto) [inline doc-deploy-stable-index/actions.yml:105]: No need to checkout any repository or branch:

```suggestion
```

**Review by jorgepiloto — CHANGES_REQUESTED** [review]

File: doc-deploy-stable-index/action.yml [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+description: |
+  This action automates the process of creating indexes and scraping the
+  stable HTML documentation artifact that contains the development version
+  of a library, then deploying it to a Meilisearch instance.
+
+inputs:
+
+  # Required inputs
+
+  scrap_cname:
```
Comment (Revathyvenugopal162) [inline doc-deploy-stable-index/action.yml:13]:
```suggestion
  cname:
```

File: doc-deploy-dev-index/action.yml [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
```
Comment (jorgepiloto) [inline doc-deploy-dev-index/action.yml:115]: The index is being passed as argument:

```suggestion
        pymeilisearch upload --template ${{ inputs.template }} --index ${{ inputs.index-name }}  --cname ${{ inputs.cname }}/version/dev/ html ${{ inputs.doc-artifact-name }}
```

**Review by jorgepiloto — CHANGES_REQUESTED** [review]
Considering that we are now accepting the index name, we could unify both actions into a single one named `doc-deploy-index`.

Also, since we are downloading artifacts, let us implement the option of `decompress-artifact`, similar to the one in `doc-deploy-stable`.

**Review by jorgepiloto — CHANGES_REQUESTED** [review]
Awesome! Could we add and example of this action in the documentation too?

**Review by jorgepiloto — APPROVED** [review]

**Pull request merged**